### PR TITLE
feat(xc-admin): add price store instructions and executor support

### PR DESCRIPTION
**Pull request by Riateche**

### CODE REVIEW

File: governance/xc_admin/packages/xc_admin_common/src/price_store.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+enum InstructionId {
+  Initialize = 0,
+  SubmitPrices = 1,
+  InitializePublisher = 2,
+}
+
+export function createPriceStoreInstruction(
+  data: PriceStoreInstruction
+): TransactionInstruction {
+  if (data.type == "Initialize") {
```
Comment (ali-bahjati) [inline governance/xc_admin/packages/xc_admin_common/src/price_store.ts:50]: i recommend using switch here, but if not triple equals (`===`)

File: governance/xc_admin/packages/xc_admin_common/src/price_store.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        payerKey: instruction.keys[0].pubkey,
+        authorityKey,
+      },
+    };
+  } else if (instructionId == InstructionId.InitializePublisher) {
+    if (instruction.data.length < 35) {
+      throw new Error("instruction data is too short");
+    }
+    const publisherKey = new PublicKey(instruction.data.subarray(3, 35));
+    if (instruction.keys.length != 5) {
```
Comment (ali-bahjati) [inline governance/xc_admin/packages/xc_admin_common/src/price_store.ts:165]: de we need to check remaining accounts to be valid?

**Review by ali-bahjati — APPROVED** [review]
Nice!

File: governance/xc_admin/packages/xc_admin_common/src/price_store.ts (hunk 2 above)
Comment (Riateche) [reply governance/xc_admin/packages/xc_admin_common/src/price_store.ts]: I've added a check that accounts and instruction data match the expected values.

File: governance/xc_admin/packages/xc_admin_common/src/price_store.ts (hunk 1 above)
Comment (Riateche) [reply governance/xc_admin/packages/xc_admin_common/src/price_store.ts]: Refactored to use switch.

**Review by ali-bahjati — APPROVED** [review]

**Pull request merged**